weblate/utils/requirements.py

```diff
@@ -169,10 +171,88 @@ def get_versions():
     return result
 
 
+def get_version_number_db(version):
+    """Returns version of database in format as (Major Version.Minor Version.Revision).
+
+    from an integer. To know more about database version,
+    refer <https://www.psycopg.org/docs/connection.html>.
+    """
```

Comment:
Can you please add tests for this? The code looks to complex for what it does and it's hard to review that it works properly. Having tests would make the verification it works correctly. It should verify all corner case you are handling (`90209` -> 9.2.9, `120003` -> 12.3, `120000` -> 12).

Maybe it would be better to avoid this postporcessing and execute `SHOW server_version;` SQL directly?

Reply:
I'll try to avoid this post-processing and string manipulations. Will add try to execute `SHOW server_version` and parse its results.

Reply:
@nijel , I think as per documentation 120000->12.0 
![image](https://user-images.githubusercontent.com/8374949/82651420-fdb19d00-9bd0-11ea-9c1a-fc3290441bb8.png)

Reply:
You're right with 12.0, these were mostly given as examples without deep investigation. Still, I think it's better to avoid such code if possible :-).

Reply:
@nijel - I have avoided the complex string manipulation that I did before. Now, I am using `connection.cursor().connection.get_server_info().split("-", 1)[0]` to parse MySQL version by quering the database and a similar approach in PostgreSQL

Reply:
Please let me if there is any better way to do things. The previous one was not definitely hackish.

Reply:
Also, after the code changes `weblate list_versions` command works fine. I have tested it after databases.